lib/helpers/webview.ts

```diff
@@ -396,12 +396,18 @@ const WebviewHelpers: WebviewHelpers = {
 
     const result: string[] = [];
     for (const {webview, pages, proc, webviewName} of webviewsMapping) {
-      if (ensureWebviewsHavePages && pages?.length === 0) {
-        logger.info(
-          `Skipping the webview '${webview}' at '${proc}' ` +
-            `since it has reported having zero pages`,
-        );
-        continue;
+      if (ensureWebviewsHavePages) {
+        if (_.isUndefined(pages)) {
```

Comment:
can pages be null or only undefined? would the `(!pages)` check be safer?

Reply:
> can pages be null

It seems that this is the only place property `pages` is substituted.

https://github.com/appium/appium-android-driver/blob/7a9f6540ec010c73e08034fbbba73b8454bed5bf/lib/helpers/webview.ts#L333-L335

If the response body from `/json/list` is parsed as `null`, `pages` can be `null`.

---

IMO, such case is "no page retrieved" situation rather than "endpoint unreachable" one which I'm adding with this PR. Do you think it's rational to handle such case in the next conditional branch (`pages.length === 0`)?

Reply:
if you say this is possible then the next branch is going to throw an exception as we cannot fetch any attribute from `null`.

Reply:
> if you say this is possible then the next branch is going to throw an exception as we cannot fetch any attribute from null.

That's right, so I meant this change.

```diff
- if (pages.length === 0) {
+ if (_.isNull(pages) || pages.length === 0) {
```

Reply:
it will work if you want to be explicit. May initial idea about `!pages?.length` was to make it simple and combine all cases into the one condition

Reply:
Thank you so much for the elaboration. I've finally got the idea and fixed as  5e6b7ebe584eb0eb088cfd8ded6dda80f9399d24.